dbt/adapters/bigquery.py

```diff
@@ -0,0 +1,300 @@
+        Creds = google.oauth2.service_account.Credentials
+
+        if method == 'oauth':
+            return None
+
+        elif method == 'service-account':
+            keyfile = config.get('keyfile')
+            return Creds.from_service_account_file(keyfile)
+
+        elif method == 'service-account-json':
+            details = config.get('config')
+            return Creds.from_service_account_info(details)
+
+        error = ('Bad `method` in profile: "{}". '
+                 'Should be "oauth" or "service-account"'.format(method))
+        raise dbt.exceptions.FailedToConnectException(error)
+
+    @classmethod
+    def get_bigquery_client(cls, config):
+        project_name = config.get('project')
+        creds = cls.get_bigquery_credentials(config)
+
+        return google.cloud.bigquery.Client(project=project_name,
+                                            credentials=creds)
+
+    @classmethod
+    def open_connection(cls, connection):
+        if connection.get('state') == 'open':
+            logger.debug('Connection is already open, skipping open.')
+            return connection
+
+        result = connection.copy()
+        credentials = connection.get('credentials', {})
+
+        try:
+            handle = cls.get_bigquery_client(credentials)
+
+        except google.auth.exceptions.DefaultCredentialsError as e:
+            logger.info("Please log into GCP to continue")
+            dbt.clients.gcloud.setup_default_credentials()
+
+            handle = cls.get_bigquery_client(credentials)
+
+        except Exception as e:
+            logger.debug("Got an error when attempting to create a bigquery "
+                         "client: '{}'".format(e))
+
+            result['handle'] = None
+            result['state'] = 'fail'
+
+            raise dbt.exceptions.FailedToConnectException(str(e))
+
+        result['handle'] = handle
+        result['state'] = 'open'
+        return result
+
+    @classmethod
+    def query_for_existing(cls, profile, schema, model_name=None):
+        dataset = cls.get_dataset(profile, schema, model_name)
+        tables = dataset.list_tables()
+
+        relation_type_lookup = {
+            'TABLE': 'table',
+            'VIEW': 'view'
+        }
+
+        existing = [(table.name, relation_type_lookup.get(table.table_type))
+                    for table in tables]
+
+        return dict(existing)
+
+    @classmethod
+    def drop_view(cls, profile, view_name, model_name):
+        schema = cls.get_default_schema(profile)
+        dataset = cls.get_dataset(profile, schema, model_name)
+        view = dataset.table(view_name)
+        view.delete()
+
+    @classmethod
+    def rename(cls, profile, from_name, to_name, model_name=None):
+        message = 'Cannot rename bigquery relation {} to {}'.format(
+                  from_name, to_name)
+        raise dbt.exceptions.NotImplementedException(message)
+
+    # hack because of current API limitations
```

Comment:
Sure, it's actually not super well documented which is part of the problem. BQ defaults to the "Legacy SQL" syntax when views are created via the API. You can set a `useLegacySql = False` flag on the `Table` object, but it looks like it's presently ignored by the API.

[Relevant issue](https://github.com/GoogleCloudPlatform/google-cloud-python/issues/3388)

The best workaround i could find was to use the `#standardSQL` header line


[Relevant issue](https://github.com/GoogleCloudPlatform/google-cloud-python/issues/3388)